bundles/org.eclipse.search.core/search/org/eclipse/search/internal/core/text/TextSearchVisitor.java

```diff
@@ -442,7 +442,12 @@ public IStatus search(TextSearchScope scope, IProgressMonitor monitor) {
 	private final IContentType TEXT_TYPE = Platform.getContentTypeManager().getContentType(IContentTypeManager.CT_TEXT);
 
 	private boolean hasBinaryContentType(IFile file) {
-		IContentType[] contentTypes = Platform.getContentTypeManager().findContentTypesFor(file.getName());
+		IContentType[] contentTypes;
+		try (java.io.InputStream contents = file.getContents()) {
+			contentTypes = Platform.getContentTypeManager().findContentTypesFor(contents, file.getName());
+		} catch (IOException | CoreException e) {
+			return false; // unknown due to failure
```

Comment:
In such case, wouldn't it be better to log it as a warning buut still resolve the contentTypes just based on the file name? That would usually lead to good enough results, wouldn't it?